File: day1/day1.js
var fs = require('fs');
var array = fs.readFileSync('input.txt').toString().split("\n");

function day1(input){
  let count1 = 0;
  let count2 = 0;
  let input1 = input.map(el => parseInt(el));
  for (let i = 0; i < input1.length; i++){
    if (i >= 1 && input1[i] > input1[i-1]){
      count1 += 1;
    }
    if (i >= 3){
      let firstSum = input1[i-3] + input1[i-2] + input1[i-1];
      let secondSum = input1[i-2] + input1[i-1] + input1[i];
      if (secondSum > firstSum){
        count2 += 1;
      }
    }
  
  }
  console.log("Part 1 " + count1);
  console.log("Part 2 " + count2);
}

day1(array);

File: day1/day1.java
import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        List<Integer> depths = new ArrayList<Integer>();
        depths = init();

       
        int[] positiveChanges = day1(depths);
        System.out.println("Part One: " + positiveChanges[1]);
        System.out.println("Part Two: " + positiveChanges[0]);

    }

    /**
     * creates a list, to which it adds all numbers from the file
     * @return a list with all numbers
     */
    public static List<Integer> init() {
        List<Integer> list = new ArrayList<Integer>();        
        try {
            FileInputStream file = new FileInputStream("input.txt");
            Scanner sc = new Scanner(file);
            while(sc.hasNextLine()) {
                list.add(Integer.parseInt(sc.nextLine()));
            }
            sc.close();
        } catch (FileNotFoundException e) {
            e.printStackTrace();
        }
        return list;
    }

    /**
     * counts the times where the following number is greater then the number before
     * @param list contains all the numbers
     * @return the number times the measurement increases
     */
    public static int[] day1(List<Integer> list) {
        int n = 0;
        int m = 0;
        for (int i = 0; i < list.size(); i++){
          if ( i >= 1 && list.get(i) > list.get(i-1)){
              n++;
          }
          if (i >= 3){
            int firstSum = list.get(i-1) + list.get(i-2) + list.get(i-3);
            int secondSum = list.get(i) + list.get(i-1) + list.get(i-2);
            if (secondSum > firstSum){
              m++;
            }
          }
        }

        int[] arr = {m, n};
        return arr;
    }
}
